pulsenotify/plugins/irc.py

```diff
@@ -95,10 +95,17 @@ def keep_alive(message, **kwargs):
 
         for chan in status_config['channels']:
             try:
-                self.irc_client.trigger(NOTIFY_SIGNAL_NAME, status=task_data.status, channel=chan,
-                                        message=status_config['message'], logs=task_data.log_data(),
-                                        task_id=task_data.id)
+                self._send_message(task_data, status_config, chan)
+            except RuntimeError as e:
+                # RuntimeError can occur when client isn't connected
+                await self.irc_client.reconnect()
```

Comment:
I think this line needs to be

```python
await self.irc_client.trigger('CLIENT_DISCONNECT')
```

The names of the functions defined in the constructor for this plugin are essentially cosmetic, they're mainly just handlers for the events named in the decorators (in this case reconnect is the handler for the CLIENT_DISCONNECT event). 

It might also be useful to add a short "sleep" here, in case the IRC client has high throughput and it's not pulse-notify that is too slow. What do you think? :)